From 85be01255fd814ce432524db3d1c95cf9becda0f Mon Sep 17 00:00:00 2001
From: mayi <3363523343@qq.com>
Date: Wed, 21 Oct 2026 18:34:08 +0800
Subject: [PATCH] fix(cvi_mpi/sample): declare GC4683 sensor modes and object

Complete the declarations referenced by existing patch 0005. Add the three linear modes, WDR mode and conditional sensor object declaration so sample_common_vi.c builds.
---
 sample/common/sensor_cfg.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sample/common/sensor_cfg.h b/sample/common/sensor_cfg.h
index 86f37c8560d38495a8256f23ba3f7a77229e3679..5d6479a5f48454495a52fea0f8bcbe986e7ab1a7 100644
--- a/sample/common/sensor_cfg.h
+++ b/sample/common/sensor_cfg.h
@@ -52,6 +52,9 @@ typedef enum _SNS_TYPE_E {
 	GCORE_GC4023_MIPI_4M_30FPS_10BIT,
 	GCORE_GC4653_MIPI_4M_30FPS_10BIT,
 	GCORE_GC4663_MIPI_4M_30FPS_10BIT,
+	GCORE_GC4683_MIPI_4M_60FPS_10BIT,
+	GCORE_GC4683_MIPI_2M_60FPS_10BIT,
+	GCORE_GC4683_MIPI_1M_120FPS_10BIT,
 	OV_OG01A1B_MIPI_2M_30FPS_10BIT,
 	OV_OG01A10_MIPI_2M_30FPS_10BIT,
 	OV_OS02G10_MIPI_2M_30FPS_10BIT,
@@ -109,6 +112,7 @@ typedef enum _SNS_TYPE_E {
 	GCORE_GC2083_MIPI_2M_30FPS_10BIT_WDR2TO1,
 	GCORE_GC2093_MIPI_2M_30FPS_10BIT_WDR2TO1,
 	GCORE_GC4663_MIPI_4M_30FPS_10BIT_WDR2TO1,
+	GCORE_GC4683_MIPI_4M_30FPS_10BIT_WDR2TO1,
 	SMS_SC285SL_MIPI_2M_30FPS_10BIT_WDR2TO1,
 	/* ------ WDR 2TO1 END ------*/
 	SNS_TYPE_WDR_BUTT,
@@ -166,6 +170,9 @@ extern ISP_SNS_OBJ_S stSnsGc4653_Obj;
 #if CONFIG_SENSOR_GCORE_GC4663
 extern ISP_SNS_OBJ_S stSnsGc4663_Obj;
 #endif
+#if CONFIG_SENSOR_GCORE_GC4683
+extern ISP_SNS_OBJ_S stSnsGc4683_Obj;
+#endif
 #if CONFIG_SENSOR_OV_OG01A1B
 extern ISP_SNS_OBJ_S stSnsOg01a1b_Obj;
 #endif
